api/src/main/java/com/datatorrent/api/SSLConfig.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.api;
+
+import java.io.Serializable;
+
+/**
+ *
+ */
+public class SSLConfig implements Serializable
```

Comment:
I am not strongly against having it in Context but feel it is unnecessary to cram this into Context. There are other attributes like AffinityRules that @sanjaypujare mentioned that are outside and potentially used as is, apart from attributes like Partitioner, StatsListener or StreamCodec that are extended. I would agree this attribute does not represent a core functionality but prefer to provide users with a more straightforward and consistent experience. I think it would have been better to have had the couple of attributes that we have in Context today outside as well. To avoid clutter at top-level a sub-package code be used.

I wouldn't block it if you choose to put it in Context.